pkg/pool-weighted/contracts/managed/ManagedPoolSettings.sol

```diff
@@ -496,6 +496,27 @@ abstract contract ManagedPoolSettings is BasePool, ProtocolFeeCache, ReentrancyG
         emit MustAllowlistLPsSet(mustAllowlistLPs);
     }
 
+    // Actual Supply
+
+    function getActualSupply() external view returns (uint256) {
+        return _getActualSupply(totalSupply());
+    }
+
+    function _getActualSupply(uint256 virtualSupply) internal view returns (uint256) {
+        if (inRecoveryMode()) {
```

Comment:
`inRecoveryMode` implicitly access `_poolState`: it should be `external` to avoid accidentally calling it like this, and we should call a version that receives the state.

(or learn to stop worrying and love global variables)

Reply:
This would a be change to `RecoveryMode` so I'm going to leave the visibility alone but change to access `_poolState` directly.